pkg/storage/storage.go

```diff
@@ -97,15 +114,43 @@ RETRYLOOP:
 				logging.Errorf("Error assigning IP: %v", err)
 				return newip, err
 			}
+			// Now check if this is allocated cluster wide
+			// When it's allocated cluster wide, we add it to a local reserved list
+			// And we try again.
+			if ipamConf.OverlappingRanges {
+				isallocated, err := clusterstore.IsAllocatedClusterWide(ctx, newip.IP)
+				if err != nil {
+					logging.Errorf("Error checking clusterwide allocation: %v", err)
+					return newip, err
+				}
+
+				if isallocated {
+					logging.Debugf("Continuing loop, IP is already allocated (possibly from another range): %v", newip)
+					// We create "dummy" records here for evaluation, but, we need to filter those out later.
+					clusterwideallocations = append(clusterwideallocations, types.IPReservation{IP: newip.IP, ContainerID: ".DuMMY."})
```

Comment:
what if `.DuMMY.` is a real containerID?

Reply:
Good catch, this is a hack! I went ahead and instead updated the `IPReservation` type to add a new property. Thanks!!